tst/EnergyPlus/unit/UnitarySystem.unit.cc

```diff
@@ -14797,9 +14797,9 @@ TEST_F(EnergyPlusFixture, Test_UnitarySystemModel_SubcoolReheatCoil)
     EXPECT_EQ(state->dataCoilCooingDX->coilCoolingDXs[0].performance.OperatingMode, 1);
     EXPECT_EQ(state->dataCoilCooingDX->coilCoolingDXs[0].performance.ModeRatio, 0.0);
     EXPECT_NEAR(thisSys->LoadSHR, 0.98533, 0.001);
-    EXPECT_NEAR(thisSys->CoilSHR, 0.97702, 0.001);
+    EXPECT_NEAR(thisSys->CoilSHR, 0.98246, 0.001);
     EXPECT_NEAR(SenOutput, -2000.0, 0.5);
-    EXPECT_NEAR(LatOutput, -346.1, 0.1);
+    EXPECT_NEAR(LatOutput, -1077.6, 0.1);
```

Comment:
This can't be right. CoilSHR = 0.98 and Sens/Total = 0.65. How can that be? I think the Tsat check is after CoilSHR is calculated, but then it wasn't correct before either at 0.852 (2000/2346.1) vs reported 0.97702. Something seems off.

Reply:
@rraustad `LatOutput` and `SenOutput` are returned by the unitary system `simulate` function, so that probably includes fan heat and reheat, if any?

Reply:
Oh, maybe that's it. The sensible to total fraction is based on delivered capacity wrt zone conditions. The coil SHR is based on coil inlet/outlet conditons. That would make sense. It depends on which reference the UnitarySystem is using. See UnitarySys::calculateCapacity. I assume this unit test is load based control. If so then this is good to go. If the UnitarySystem was referencing it's inlet node then I would think the SHRs would be much closer.

        if (this->m_ControlType == ControlType::Setpoint) {
            RefTemp = state.dataLoopNodes->Node(this->AirInNode).Temp;
            RefHumRat = state.dataLoopNodes->Node(this->AirInNode).HumRat;
        } else {
            RefTemp = state.dataLoopNodes->Node(this->NodeNumOfControlledZone).Temp;
            RefHumRat = state.dataLoopNodes->Node(this->NodeNumOfControlledZone).HumRat;
        }

Reply:
@rraustad It turns out that the calculations in UnitarySystem are actually based on delivered capacity wrt zone conditions. I added some new checks to the unit tests and they don't line up. If you agree, I'll post a new issue about CoilSHR, clean up the unit tests and then we can get this saturation check merged in.

Reply:
Which new checks in unit tests? I thought the Tsat correction for the coil model (not the ones for bypass mixing or contfan) was after the point where CoilSHR is calculated. If we move the coil outlet condition the CoilSHR will change. Does the code make that correction?

Reply:
Sorry - bad timing. I just pushed a new commit https://github.com/NREL/EnergyPlus/pull/9187/commits/dbaea50c3d23da4de0eb52c0fa1c81f8725c5ed1 that adds checks for tOut <= tSat in the subcool reheat unit tests.  If you search through UnitarySystem for `CoilSHR`, you'll see a variety of methods from assuming it equals `LoadSHR` to calculating it, to doing a plr or speed ratio adjustment. So, I think that all warrants some review, but I'd like that to be a separate effort from this.

Reply:
I saw those new lines but thought you added something to identify issues with CoilSHR. I also know about how UnitarySystem is manipulating the SHR passed to the coil model. So yes, a good review of that would be helpful. Since UnitarySystem is calculating this it seems like CoilSHR is a target rather than an indication of coil performance.